### PR TITLE
Replace mockfs tests

### PR DESCRIPTION
**Pull request by cindycs212**
#### :heavy_check_mark: Checklist
@awanlin - Can you please review? Thanks!

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -56,6 +56,7 @@
   },
   "version": "1.19.0",
   "dependencies": {
+    "@backstage/backend-test-utils": "workspace:^",
```
Comment (awanlin) [inline package.json:59]: I'm not sure why this would be needed? Pretty sure you can removed this

File: .changeset/many-flowers-buy.md
Comment (awanlin) [inline .changeset/many-flowers-buy.md:1]: As you are only changing tests you don't need a changeset, just delete this

**Review by awanlin — CHANGES_REQUESTED** [review]
Thanks for this contribution @cindycs212! 🚀 

The overall code changes look good to me, just left a couple of small comments

File: package.json (hunk 1 above)
Comment (cindycs212) [reply package.json:59]: removed this and the changeset :)

Comment (awanlin) [reply package.json:59]: It still looks to be there from my end 🤔

Comment (awanlin) [reply package.json:59]: Also, make sure that the `yarn.lock` change is not there as well, pretty sure these two things are what are causing the builds to fail

Comment (awanlin) [reply package.json:59]: In the end you should only have two files with changes: `confluenceToMarkdown.examples.test.ts` and ``confluenceToMarkdown.test.ts`

**Comment by awanlin** [comment]
> Thanks for the contribution! All commits need to be DCO signed before they are reviewed. Please refer to the the [DCO section in CONTRIBUTING.md](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin) or the [DCO](https://github.com/backstage/backstage/runs/17939887050) status for more info.

@cindycs212, you need to make sure to sign your commits, there's details in the quote above ☝️

**Comment by cindycs212** [comment]
> > Thanks for the contribution! All commits need to be DCO signed before they are reviewed. Please refer to the the [DCO section in CONTRIBUTING.md](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin) or the [DCO](https://github.com/backstage/backstage/runs/17939887050) status for more info.
> 
> @cindycs212, you need to make sure to sign your commits, there's details in the quote above ☝️

Sign offs should be done now :) I used git rebase -i --signoff $(git merge-base -a master HEAD) from the DCO section

File: package.json (hunk 1 above)
Comment (cindycs212) [reply package.json]: Removed from both package.json and yarn.lock :) The two confluenceToMarkdown tests are the only files changed now, thanks so much Andrew!!

**Comment by cindycs212** [comment]
> OK, looks like the first two commits are not signed so those need to be fixed. Also, the tests are failing in the build, are these passing for you locally?

I've signed off all the commits this time :) but I'm getting the following build error and not sure how to proceed, do you have suggestions? Thank you!

```
Error: Failed to compile '../../node_modules/__backstage-autodetected-plugins__.js':
  Module parse failed: Unexpected token (1:8115)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
> window['__@backstage/discovered__'] = { modules: [{ name: "@backstage/plugin-adr", export: undefined, default: require('@backstage/plugin-adr').default },{ name: "@backstage/plugin-adr", export: "./alpha", default: require('@backstage/plugin-adr/alpha').default },{ name: "@backstage/plugin-airbrake", export: undefined, default: require('@backstage/plugin-airbrake').default },{ name: "@backstage/plugin-apache-airflow", export: undefined, default: require('@backstage/plugin-apache-airflow').default },{ name: "@backstage/plugin-api-docs", export: undefined, default: require('@backstage/plugin-api-docs').default },{ name: "@backstage/plugin-azure-devops", export: undefined, default: require('@backstage/plugin-azure-devops').default },{ name: "@backstage/plugin-azure-sites", export: undefined, default: require('@backstage/plugin-azure-sites').default },{ name: "@backstage/plugin-badges", export: undefined, default: require('@backstage/plugin-badges').default },{ name: "@backstage/plugin-catalog", export: undefined, default: require('@backstage/plugin-catalog').default },{ name: "@backstage/plugin-catalog", export: "./alpha", default: require('@backstage/plugin-catalog/alpha').default },{ name: "@backstage/plugin-catalog-graph", export: undefined, default: require('@backstage/plugin-catalog-graph').default },{ name: "@backstage/plugin-catalog-import", export: undefined, default: require('@backstage/plugin-catalog-import').default },{ name: "@backstage/plugin-catalog-import", export: "./alpha", default: require('@backstage/plugin-catalog-import/alpha').default },{ name: "@backstage/plugin-catalog-unprocessed-entities", export: undefined, default: require('@backstage/plugin-catalog-unprocessed-entities').default },{ name: "@backstage/plugin-circleci", export: undefined, default: require('@backstage/plugin-circleci').default },{ name: "@backstage/plugin-cloudbuild", export: undefined, default: require('@backstage/plugin-cloudbuild').default },{ name: "@backstage/plugin-code-coverage", export: undefined, default: require('@backstage/plugin-code-coverage').default },{ name: "@backstage/plugin-cost-insights", export: undefined, default: require('@backstage/plugin-cost-insights').default },{ name: "@backstage/plugin-devtools", export: undefined, default: require('@backstage/plugin-devtools').default },{ name: "@backstage/plugin-dynatrace", export: undefined, default: require('@backstage/plugin-dynatrace').default },{ name: "@backstage/plugin-entity-feedback", export: undefined, default: require('@backstage/plugin-entity-feedback').default },{ name: "@backstage/plugin-explore", export: undefined, default: require('@backstage/plugin-explore').default },{ name: "@backstage/plugin-explore", export: "./alpha", default: require('@backstage/plugin-explore/alpha').default },{ name: "@backstage/plugin-gcalendar", export: undefined, default: require('@backstage/plugin-gcalendar').default },{ name: "@backstage/plugin-gcp-projects", export: undefined, default: require('@backstage/plugin-gcp-projects').default },{ name: "@backstage/plugin-github-actions", export: undefined, default: require('@backstage/plugin-github-actions').default },{ name: "@backstage/plugin-gocd", export: undefined, default: require('@backstage/plugin-gocd').default },{ name: "@backstage/plugin-graphiql", export: undefined, default: require('@backstage/plugin-graphiql').default },{ name: "@backstage/plugin-graphiql", export: "./alpha", default: require('@backstage/plugin-graphiql/alpha').default },{ name: "@backstage/plugin-home", export: undefined, default: require('@backstage/plugin-home').default },{ name: "@backstage/plugin-home", export: "./alpha", default: require('@backstage/plugin-home/alpha').default },{ name: "@backstage/plugin-jenkins", export: undefined, default: require('@backstage/plugin-jenkins').default },{ name: "@backstage/plugin-kafka", export: undefined, default: require('@backstage/plugin-kafka').default },{ name: "@backstage/plugin-kubernetes", export: undefined, default: require('@backstage/plugin-kubernetes').default },{ name: "@backstage/plugin-lighthouse", export: undefined, default: require('@backstage/plugin-lighthouse').default },{ name: "@backstage/plugin-linguist", export: undefined, default: require('@backstage/plugin-linguist').default },{ name: "@backstage/plugin-microsoft-calendar", export: undefined, default: require('@backstage/plugin-microsoft-calendar').default },{ name: "@backstage/plugin-newrelic", export: undefined, default: require('@backstage/plugin-newrelic').default },{ name: "@backstage/plugin-newrelic-dashboard", export: undefined, default: require('@backstage/plugin-newrelic-dashboard').default },{ name: "@backstage/plugin-octopus-deploy", export: undefined, default: require('@backstage/plugin-octopus-deploy').default },{ name: "@backstage/plugin-org", export: undefined, default: require('@backstage/plugin-org').default },{ name: "@backstage/plugin-pagerduty", export: undefined, default: require('@backstage/plugin-pagerduty').default },{ name: "@backstage/plugin-playlist", export: undefined, default: require('@backstage/plugin-playlist').default },{ name: "@backstage/plugin-puppetdb", export: undefined, default: require('@backstage/plugin-puppetdb').default },{ name: "@backstage/plugin-rollbar", export: undefined, default: require('@backstage/plugin-rollbar').default },{ name: "@backstage/plugin-scaffolder", export: undefined, default: require('@backstage/plugin-scaffolder').default },{ name: "@backstage/plugin-scaffolder", export: "./alpha", default: require('@backstage/plugin-scaffolder/alpha').default },{ name: "@backstage/plugin-search", export: undefined, default: require('@backstage/plugin-search').default },{ name: "@backstage/plugin-search", export: "./alpha", default: require('@backstage/plugin-search/alpha').default },{ name: "@backstage/plugin-sentry", export: undefined, default: require('@backstage/plugin-sentry').default },{ name: "@backstage/plugin-shortcuts", export: undefined, default: require('@backstage/plugin-shortcuts').default },{ name: "@backstage/plugin-stackstorm", export: undefined, default: require('@backstage/plugin-stackstorm').default },{ name: "@backstage/plugin-tech-insights", export: undefined, default: require('@backstage/plugin-tech-insights').default },{ name: "@backstage/plugin-tech-radar", export: undefined, default: require('@backstage/plugin-tech-radar').default },{ name: "@backstage/plugin-tech-radar", export: "./alpha", default: require('@backstage/plugin-tech-radar/alpha').default },{ name: "@backstage/plugin-techdocs", export: undefined, default: require('@backstage/plugin-techdocs').default },{ name: "@backstage/plugin-techdocs", export: "./alpha", default: require('@backstage/plugin-techdocs/alpha').default },{ name: "@backstage/plugin-techdocs-module-addons-contrib", export: undefined, default: require('@backstage/plugin-techdocs-module-addons-contrib').default },{ name: "@backstage/plugin-todo", export: undefined, default: require('@backstage/plugin-todo').default },{ name: "@backstage/plugin-user-settings", export: undefined, default: require('@backstage/plugin-user-settings').default },{ name: "@backstage/plugin-user-settings", export: "./alpha", default: require('@backstage/plugin-user-settings/alpha').default },{ name: "@oriflame/backstage-plugin-score-card", export: undefined, default: require('@oriflame/backstage-plugin-score-card').default },{ name: "@roadiehq/backstage-plugin-buildkite", export: undefined, default: require('@roadiehq/backstage-plugin-buildkite').default },{ name: "@roadiehq/backstage-plugin-github-insights", export: undefined, default: require('@roadiehq/backstage-plugin-github-insights').default },{ name: "@roadiehq/backstage-plugin-github-pull-requests", export: undefined, default: require('@roadiehq/backstage-plugin-github-pull-requests').default },{ name: "@roadiehq/backstage-plugin-travis-ci", export: undefined, default: require('@roadiehq/backstage-plugin-travis-ci').default },{ name: "app-next-example-plugin", export: undefined, default: require('app-next-example-plugin').default }] };ghts", export: undefined, default: require('@roadiehq/backstage-plugin-github-insights').default },{ name: "@roadiehq/backstage-plugin-github-pull-requests", export: undefined, default: require('@roadiehq/backstage-plugin-github-pull-requests').default },{ name: "@roadiehq/backstage-plugin-travis-ci", export: undefined, default: require('@roadiehq/backstage-plugin-travis-ci').default }] };
```

**Comment by freben** [comment]
Ping @cindycs212 - Hi! Did you get a chance to try out the commented solution?

**Comment by freben** [comment]
I just rebased this and removed the module mocks. Still need to adjust the tests themselves to stop failing.

**Comment by tradcliffe-expediagroup** [comment]
Hi @freben sorry for the late reply, I am just getting back from holiday. I would be happy to take a look next week!

**Comment by freben** [comment]
Nice! Reopened for now.

**Comment by tradcliffe-expediagroup** [comment]
@cindycs212 Could I get permission to push to your fork please? I have the changes done locally, but do not have permission to push. Thank you.

**Comment by freben** [comment]
@tradcliffe-expediagroup since they DCO signed their contributions and hence donated them to the project, it's within your rights to make a new branch / fork with these contents and submit them as your own new PR. Feel free to do so at your convenience, if you don't get a timely response.

Thanks for your work on this!

**Comment by freben** [comment]
Closed by #22043

**Pull request closed without merging**